### PR TITLE
Add MailHog and Dada Mail; Merge Mailing Lists and Newsletters

### PR DESCRIPTION
**Pull request by kohenkatz**

### CODE REVIEW

**Comment by kohenkatz** [comment]
Truth is, I'm not sure whether Dada Mail should be in the "Mailing list servers" category or in the "Newsletter management" category, since it does both of those things.  Should it be in both?

**Comment by nodiscc** [comment]
I agree, maybe we should merge both sections?


#### Mailing lists and Newsletters

_Mailing lists servers and mass mailing software - one message to many recipients_

**Comment by Kickball** [comment]
@nodiscc @AndyR207 @kohenkatz 

I'm happy to merge mailing lists and newsletters, they're similar enough.

**Comment by kohenkatz** [comment]
Updated PR

File: README.md [hunk 1]
```diff
@@ -237,20 +237,16 @@ _MDAs - IMAP/POP3 software_
 * [Dovecot](http://www.dovecot.org/) - IMAP and POP3 server written primarily with security in mind. ([Source Code](https://github.com/dovecot/core)) `MIT/LGPLv2` `C`
 * [Piler](http://www.mailpiler.org/wiki/start) - feature rich open source email archiving solution. ([Source Code](https://bitbucket.org/jsuto/piler)) `GPLv3` `C`
 
-#### Mailing lists
+#### Mailing lists and Newsletters
 
-_Mailing lists servers._
+_Mailing lists servers and mass mailing software - one message to many recipients._
 
+  * [Dada Mail](http://dadamailproject.com/) - A web-based list management system that can be used for announcement lists and/or discussion lists. (Also has a "Pro" version which allows for more/larger mailing lists and branding removal.) ([Source Code](https://github.com/justingit/dada-mail)) `GPLv2` `Perl`
```
Comment (AndyR207) [inline README.md]: You need the period outside the bracket otherwise our scanner will flag it as invalid.

**Comment by AndyR207** [comment]
I'm happy with the merge of the sections.

File: README.md (hunk 1 above)
Comment (kohenkatz) [inline README.md]: Can we get the scanner to recognize proper grammar instead?

Instead, I think I'll just remove the parenthetical - it's not really necessary.

File: README.md [hunk 2]
```diff
@@ -21,8 +21,7 @@ See [Contributing](#contributing).
       - [Complete solutions](#complete-solutions)
       - [Mail Transfer Agents](#mail-transfer-agents)
       - [Mail Delivery Agents](#mail-delivery-agents)
-      - [Mailing lists](#mailing-lists)
-      - [Newsletter management](#newsletter-management)
+      - [Mailing lists and newsletter management](#mailing-lists)
```
Comment (nodiscc) [inline README.md]: - [Mailing lists and newsletters](#mailing-lists-and-newsletters)

Comment (kohenkatz) [inline README.md]: Fixed

**Comment by nodiscc** [comment]
Thanks

**Pull request merged**